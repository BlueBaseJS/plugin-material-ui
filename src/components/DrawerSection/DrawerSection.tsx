import { DrawerSectionProps } from '@bluebase/components';
import List from '@material-ui/core/List';
import ListSubheader from '@material-ui/core/ListSubheader';
import React from 'react';
import { componentMapper } from '@bluebase/component-mapper';

export const DrawerSection = componentMapper<DrawerSectionProps>(List, {
	children: 'children',
	subheader: ({ title }: DrawerSectionProps) => title ? <ListSubheader>{title}</ListSubheader> : null
});

