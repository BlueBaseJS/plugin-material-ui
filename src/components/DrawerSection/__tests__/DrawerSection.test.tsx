import { shallow } from 'enzyme';
import React from 'react';

import { DrawerSection } from '../DrawerSection';

describe('DrawerSection', () => {

	test('should forward title prop as subheader', () => {
		const component = shallow(
			<DrawerSection title="Item 1" />
		);
		expect(component.props().subheader).toBeTruthy();
	});

	test('should forward title prop as subheader=null', () => {
		const component = shallow(
			<DrawerSection />
		);
		// expect(component).toMatchSnapshot();
		expect(component.props().subheader).toBeNull();
	});

});
