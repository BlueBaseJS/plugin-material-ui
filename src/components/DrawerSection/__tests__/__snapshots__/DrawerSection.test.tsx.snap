// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerSection should forward title prop as subheader=null 1`] = `
<WithStyles(List)
  subheader={null}
/>
`;
